### PR TITLE
Make Azure driver consistent and add it to the factory

### PR DESCRIPTION
**Pull request by squeedee**
Make Azure driver consistent and add it to the factory

* added to the factory
* integration tests only require a token with access to a project the tests can create and delete repos
* azure driver does not require the project up front. Instead org/proj are encoded in the namespace
* CreateRef returns a ref object
* Link in PR points to web endpoint.
* Repo.List moved to Repo.ListOrganisation as that more accurately fits the results.
* Removed hook methods until they can be re-implemented to fit the per-repo model (AZDO projects can have multiple repositories, and the hooks are per-project)
* Properly handle list options on the Repository.List method (jx-scm gets into an infinite loop otherwise)
* Include the Clone And CloneSSH links in the repository object

### CODE REVIEW

File: scm/driver/azure/repo.go [hunk 1]
```diff
@@ -35,7 +56,32 @@ func (s *RepositoryService) FindCombinedStatus(ctx context.Context, repo, ref st
...
+			ID    string `json:"id"`
+			Name  string `json:"name"`
+			State string `json:"state"`
+			URL   string `json:"url"`
+		}{
+			ID: proj.ID,
+		},
+	}
+	out := new(repository)
+	endpoint := fmt.Sprintf("https://dev.azure.com/%s/%s/_apis/git/repositories?api-version=6.0", ro.org, ro.project)
```
Comment (garethjevans) [inline scm/driver/azure/repo.go:81]: should this be hardcoded to dev.azure.com?

File: scm/driver/azure/repo.go [hunk 2]
```diff
@@ -238,106 +212,61 @@ type repository struct {
...
+		Link:      from.WebURL,
+		Clone:     from.RemoteURL,
+		CloneSSH:  from.SSHURL,
+		Branch:    scm.TrimRef(from.DefaultBranch),
 	}
+}
+
+func (s *RepositoryService) getProject(ctx context.Context, ro *repoObj) (*project, *scm.Response, error) {
+	proj := new(project)
+	projectEndpoint := fmt.Sprintf("https://dev.azure.com/%s/_apis/projects/%s?api-version=6.0", ro.org, ro.project)
```
Comment (garethjevans) [inline scm/driver/azure/repo.go:268]: hardcoding to dev.azure.com?

File: scm/driver/azure/repo.go (hunk 1 above)
Comment (squeedee) [reply scm/driver/azure/repo.go]: thanks for the catch

**Pull request merged**